modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/QueryMemoryManager.java

```diff
@@ -325,6 +326,15 @@ public long memoryLimit() {
         return globalQuota;
     }
 
+    /**
+     *
+     * @return memory in bytes currently left available for the queries on this node.
+     */
+    public long getFreeMem() {
+        return ((LongMetric)ctx.metric().registry(SqlMemoryStatisticsHolder.SQL_QUOTAS_REG_NAME).findMetric("freeMem")).value();
+    }
```

Comment:
Why the QueryMemoryManager refers to the MetricsRegistry?
QueryMemoryManager is a metric supplier already and is able to calculate the metric value directly.

Reply:
You are right, the actual tracking is done by QueryMemoryMAnager

freeMem is a registration with GridMetricsManager 

This code, and subsequent tests were written a while back.
see: https://github.com/gridgain/gridgain/commit/e6adea02b6463c268d3c67bb973e9dbf754a8ff1#diff-eb122bf3d20dca6a87012b827be3303b6ca6c5fba323c2fc9ff01a1de2449075R68

The current code uses this existing structure, taking advantage of the "freeMem" registration, along w/new tests for the new jmx property.

This prevents code duplication in case the freeMem tracking needs to change in the future(as it has many times in the past)

Reply:
@akorensh Please, never use `MetricRegistry.find()` in production code.